pkg/apis/eventing/v1alpha1/trigger_lifecycle.go

```diff
@@ -85,3 +88,28 @@ func (ts *TriggerStatus) MarkNotSubscribed(reason, messageFormat string, message
 func (ts *TriggerStatus) MarkSubscriptionNotOwned(sub *messagingv1alpha1.Subscription) {
 	triggerCondSet.Manage(ts).MarkFalse(TriggerConditionSubscribed, "SubscriptionNotOwned", "Subscription %q is not owned by this Trigger.", sub.Name)
 }
+
+func (ts *TriggerStatus) MarkDependencyReadySucceeded() {
+	triggerCondSet.Manage(ts).MarkTrue(TriggerConditionDependencyReady)
+}
+
+func (ts *TriggerStatus) MarkDependencyReadyFailed(reason, messageFormat string, messageA ...interface{}) {
+	triggerCondSet.Manage(ts).MarkFalse(TriggerConditionDependencyReady, reason, messageFormat, messageA...)
+}
+
+func (ts *TriggerStatus) MarkDependencyReadyUnknown(reason, messageFormat string, messageA ...interface{}) {
+	triggerCondSet.Manage(ts).MarkUnknown(TriggerConditionDependencyReady, reason, messageFormat, messageA...)
+}
+
+func (ts *TriggerStatus) PropagateDependencyStatus(ks *duckv1alpha1.KResource) {
+	kc := ks.Status.GetCondition(duckv1alpha1.ConditionReady)
+	if kc != nil && kc.IsTrue() {
+		ts.MarkDependencyReadySucceeded()
+	} else {
+		msg := "nil"
```

Comment:
I suggest using `unknown` as the default string here. It might make more sense to users.

Reply:
I followed the code style from other resource lifecycle. Should we change all these to default "unknown"

Reply:
Ok, if other resources are using `nil`, ignore my comment. Better to be consistent. I'll create an issue to consider whether this should be changed for all resources.

Reply:
https://github.com/knative/eventing/issues/1861